### ISSUE TITLE
Aborting a request not working ?

### ISSUE THREAD

**Issue by vdaubry** [typhoeus/typhoeus#466]
Returning ``` :abort ``` from ``` on_body ``` doesn't cancel the request.

I expect this code to return immediatly :

```
require 'typhoeus'

url = "http://imgsv.imaging.nikon.com/lineup/dslr/d600/img/sample01/img_04_l.jpg"
request = Typhoeus::Request.new(url)
request.on_body do |chunk|
  :abort
end
request.run
```

But it runs untill the image is completely downloaded. Am i missing something obvious ?

PS: I read #423 and [typhoeus/ethon@6117872](https://github.com/typhoeus/ethon/commit/61178720)

**Comment by Zapotek** [comment]
Seems to be working when requesting something like https://example.com but not when grabbing that image.

**Comment by Zapotek** [comment]
And it seems to be working with images in general, like for example: https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/SIPI_Jelly_Beans_4.1.07.tiff/lossy-page1-256px-SIPI_Jelly_Beans_4.1.07.tiff.jpg

**Comment by vdaubry** [comment]
This image is very large (12.5mb) so the problem was easy to verify. I think you have the feeling it works because the image you use is small.

In the same way i would expect this code to print "received bytes" only once. But it seems to receive all the file.

```

require 'typhoeus'
url = "https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/SIPI_Jelly_Beans_4.1.07.tiff/lossy-page1-256px-SIPI_Jelly_Beans_4.1.07.tiff.jpg"
request = Typhoeus::Request.new(url)
request.on_body do |chunk|
  puts "received bytes"
  :abort
end
request.run

```

If i understand correctly returning ``` :abort ``` should trigger a ``` CURLE_WRITE_ERROR ``` , is there a way to see if this error is raised ?

**Comment by Zapotek** [comment]
I was checking the response body which was reported as empty.
I just now remembered that there is no body assigned when you stream the response; I've basically wasted your time, carry on...

**Comment by i0rek** [comment]
@vdaubry thanks for reporting this issue! @Zapotek thanks for chiming in and helping!
Sorry for the late response - I was on vacation.

@vdaubry if you change your test script to something like the following you will be able to see the error and a bunch of other stuff:

```ruby
require 'typhoeus'

Typhoeus::Config.verbose=true
url = "http://imgsv.imaging.nikon.com/lineup/dslr/d600/img/sample01/img_04_l.jpg"
request = Typhoeus::Request.new(url)
request.on_body do
  puts "abort!"
  :abort
end
request.run
puts request.response.return_code
```

Thanks to your report I was able to identify the issue and provide a fix: https://github.com/typhoeus/typhoeus/pull/468. If you confirm it works I will merge it.

**Comment by vdaubry** [comment]
Yes ! It's working. Good to merge :)

Thanks a lot for responding to this issue.

**Comment by i0rek** [comment]
I released `0.7.3`.
